harfbuzz_overflow_resolution.md

```diff
@@ -0,0 +1,157 @@
+# Introduction
+
+Several tables in the opentype format are formed internally by a graph of subtables. Parent node's
+reference their children through the use of positive offsets, which are typically 16 bits wide.
+For storage in the font file the graph must be given a topological ordering and then the subtables
+packed in serial according to that ordering. Since 16 bit offsets have a maximum value of 65,535 if
+the distance between a parent subtable and a child is more then 65,535 bytes then it's not possible
+for the offset to encode that edge.
+
+For many fonts with complex layouts it's not unusual for the GSUB/GPOS table to be larger than 65kb.
+As a result these types of fonts are susceptible to offset overflows when serializing to the binary
+font format.
+
+Offset overflows can be resolved through a variety of strategies:
+*  Often times a different topological ordering can avoid the overflows.
+*  In the graph subtables can have many parents. Which can result in the link from furthest parent(s)
+   being at risk for overflows. In these cases it's possible to duplicate the shared subtable which
```

Comment:
It's weird to enter a bullet about how to resolve overflow by presenting how it can occur. Maybe we should have two lists, Overflows can occur due to: ... and Overflows can be resolved by .... Or a combined one that more clearly explains both points?